Vic2ToHoI4/Source/HOI4World/Decisions/DecisionsInCategory.cpp

```diff
@@ -0,0 +1,222 @@
+#include "DecisionsInCategory.h"
+#include "../Events/Events.h"
+#include "DecisionsInCategory.h"
+
+
+HoI4::DecisionsInCategory::DecisionsInCategory(std::string categoryName, std::istream& theStream):
+	 name(std::move(categoryName))
+{
+	registerKeyword(std::regex("[A-Za-z0-9_]+"), [this](const std::string& decisionName, std::istream& theStream) {
+		const decision theDecision(decisionName, theStream);
+		theDecisions.push_back(theDecision);
+	});
+
+	parseStream(theStream);
+}
+
+
+void updateOpenUpPoliticalDiscourse(HoI4::decision& decisionToUpdate, const std::set<std::string>& majorIdeologies);
+void updateDiscreditGovernment(HoI4::decision& decisionToUpdate, const std::set<std::string>& majorIdeologies);
+void updateInstitutePressCensorship(HoI4::decision& decisionToUpdate, const std::set<std::string>& majorIdeologies);
+void updateIgniteTheIdeologyCivilWar(HoI4::decision& decisionToUpdate, const std::set<std::string>& majorIdeologies);
+
+void HoI4::DecisionsInCategory::updatePoliticalDecisions(const std::set<std::string>& majorIdeologies,
+	 const Events& theEvents)
+{
+	for (auto& theDecision: theDecisions)
+	{
+		const auto& decisionName = theDecision.getName();
+		if (decisionName.substr(0, 28) == "open_up_political_discourse_")
```

Comment:
a simpler syntax is  
if (decisionName.find("open_up_political_discourse_") == 0) 
it returns 0 if found at position 0, which you're checking.